packages/scheduler/src/forks/SchedulerHostConfig.default.js

```diff
@@ -244,103 +207,162 @@ if (
           hasTimeRemaining,
           currentTime,
         );
-        if (hasMoreWork) {
-          // Ensure the next frame is scheduled.
-          if (!isAnimationFrameScheduled) {
-            isAnimationFrameScheduled = true;
-            requestAnimationFrameWithTimeout(animationTick);
-          }
-        } else {
+        if (!hasMoreWork) {
           scheduledHostCallback = null;
         }
       } catch (error) {
         // If a scheduler task throws, exit the current browser task so the
         // error can be observed, and post a new task as soon as possible
         // so we can continue where we left off.
-        isMessageEventScheduled = true;
-        port.postMessage(undefined);
+        port.postMessage(null);
         throw error;
       }
-      // Yielding to the browser will give it a chance to paint, so we can
-      // reset this.
-      needsPaint = false;
     }
+    // Yielding to the browser will give it a chance to paint, so we can
+    // reset this.
+    needsPaint = false;
```

Comment:
Is this still valid in the event that `performWorkUntilDeadline` gets called by timeout or rIC?

Reply:
Not in that case but it is relevant in the case where `forceFrameRate` sets the frame length to an artificially large size. We should track the chosen frame length and the hardware frame length separately. Planning to do that in a follow up.